lib/chef/knife/client_create.rb

```diff
@@ -81,6 +81,15 @@ def run
           client.public_key File.read(File.expand_path(config[:public_key]))
         end
 
+        # Check the file before creating the client so the api is more transactional.
+        if config[:file]
+          file = config[:file]
+          unless File.exist?(file) ? File.writable?(file) : File.writable?(File.dirname(file))
+            ui.fatal "File #{config[:file]} is not writable.  Check permissions."
```

Comment:
This should really be two different conditionals with two different error messages.  When the file does not exist and the directory is not writable the error message should be separate and there should really be two different conditionals here for readability.